afl-fuzz.c

```diff
@@ -824,7 +824,14 @@ static void add_to_queue(u8* fname, u32 len, u8 passed_det) {
 
   cycles_wo_finds = 0;
 
-  if (!(queued_paths % 100)) {
+  /* we set here next_100 pointer on every new entry
+   * which index is multiple of 100 and positive
+   * to get faster entry with corresponding big index (>=100)
+   * in splicing stage
+   *
+   * note that while index is 100 (e.g.), queued_paths is 101
+   */
+  if (queued_paths % 100 == 1 && queued_paths > 1) {
```

Comment:
Thanks! I think I see now what the issue is: I think we should set `next_100` for index 0, 100, 200, etc. But we were setting it for 0, 99, 199, etc. I.e., we should consider the index in this expression, not the number of entries.

If that's correct, would changing this line to `if ((queued_paths - 1) % 100 == 0)` work?

The problem with using `if (queued_paths % 100 == 1 && queued_paths > 1)` is that we're not going to set `next_100` for the first element (index 0).

If my above understanding/suggestion works, the comment could just say something like:

`/* Set next_100 pointer for every 100th element (index 0, 100, etc) to allow faster iteration. */`

Reply:
Yes, it's correct!

With a little amendment. When index is 100, we set `next_100` for the element `q_prev100`, which is exactly the first element (index 0). And we can't set `next_100`, while we have the only element in queue, so we have to check that `queued_paths > 1`.

Could you explain please why `(queued_paths - 1) % 100 == 0` is better than `queued_paths % 100 == 1`?

Reply:
> With a little amendment. When index is 100, we set `next_100` for the element `q_prev100`, which is exactly the first element (index 0). And we can't set `next_100`, while we have the only element in queue, so we have to check that `queued_paths > 1`.

Ah right, got it, thanks!

> Could you explain please why `(queued_paths - 1) % 100 == 0` is better than `queued_paths % 100 == 1`?

It's the same, just wanted to capture better than we want `index % 100 == 0`, but leaving it up to you.

Thanks again for the fix, and sorry for the delay with the review!

Reply:
All right. Thank you!